src/sentry/api/endpoints/group_stats.py

```diff
@@ -3,14 +3,27 @@
 from rest_framework.response import Response
 
 from sentry import tsdb
-from sentry.api.base import StatsMixin
+from sentry.api.base import EnvironmentMixin, StatsMixin
 from sentry.api.bases.group import GroupEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.models import Environment
 
 
-class GroupStatsEndpoint(GroupEndpoint, StatsMixin):
+class GroupStatsEndpoint(GroupEndpoint, EnvironmentMixin, StatsMixin):
     def get(self, request, group):
+        try:
+            environment_id = self._get_environment_id_from_request(
+                request,
+                group.project.organization_id,
+            )
+        except Environment.DoesNotExist:
+            raise ResourceDoesNotExist
```

Comment:
I'm conflicted on whether this should actually raise or just return an empty series, but for now I'm just following the pattern established in GH-6498 for consistency.

Reply:
Worth noting that in some places I do return an empty collection. Typically when the environment specific bits were a subset of the response. But if the entire object being returned itself is part of an environment scope I felt like 404 was the only proper response if the environment didn't even exist. Totally open to pushback, though.

Reply:
https://github.com/getsentry/sentry/blob/4eaef02e01cfa883b97388b59a24ecf3a7140334/src/sentry/api/endpoints/group_details.py#L218

Reply:
Also: there's no quick way to return an empty series from TSDB (without just providing a known invalid key like -1 and hitting the database anyway.) I'm going to leave this as-is for now since I think it's easier to loosen the constraint later than add it if we decide it's necessary.